x-pack/plugins/security_solution/server/endpoint/services/actions/utils.ts

```diff
@@ -127,7 +137,7 @@ export const getActionCompletionInfo = (
       if (!normalizedAgentResponse.wasSuccessful) {
         completedInfo.wasSuccessful = false;
         responseErrors.push(
-          ...(normalizedAgentResponse.errors ? normalizedAgentResponse.errors : [])
+          ...(normalizedAgentResponse.errors ? normalizedAgentResponse.errors : '')
```

Comment:
Ok. I did not realize this also worked (spreading an empty string, will convert that string to an array first by `.split()` and will then spread that, and since the conversion to array resulted in `[]` is just work the same as defining an empty array. You made me do console.log tests LOL


now - that being said - is setting this a string obvious to a dev as to what is actually happening?

I'm ok leaving it as you have, but just wanted to point out that not everyone might actually understand how spread of a string works.

Reply:
So I did this because the `responseErrors` is either `undefined` or is a `string[]`. It is not an array of arrays.
So either we push an error string when it exists or we push an empty string. 

You're still spreading the array when you have `['']`.  You are correct in pointing out that you can't spread a string.

What I'm wondering now is why it allows you to push an empty array inside into a string array 🤔 .

Reply:
At this point here, `responseErrors` is an Array (look at line 127). It's never `undefined`. I would suggest you revert this back to `[]`.